.github/workflows/quality-checks.yml

```
@@ -9,15 +9,19 @@ jobs:
 
     strategy:
       matrix:
-        python-version:
-          - "3.12"
-          - "3.11"
-          - "3.10"
-          - "3.9"
+        python-version: ["3.12", "3.11", "3.10", "3.9"]
 
     steps:
       - uses: actions/checkout@v4
 
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
```

How does this install the correct version?

For my old mind, can we have this too...
```suggestion
        run: python -c "import sys; print(sys.version)"
```
```suggestion
        run: |
          python -c "import sys; print(sys.version)"
          python --version
```

---

I'm happy to give it a shot, however the changes I've made are as per GitHub's official documentation which I have linked in the issue's description

---

Either way, we need one only between `--version` and `-c "import..."` as they'd do the same thing. Considering using the `-c` flag is Git's official recommendation, I would maybe stick with it, or get rid of lines 22-23 alltogether as it's just a verbose way to display the correct python version in use (which is going to show up muddled somewhere atop each pytest run)

---

It may be unfounded, but I have more confidence that `python --version` will display the version that will be used in the following steps.

I do prefer to explicitly have this output.